File: crawler/src/model/artist.js
class Artist {
    /**
     * @param {string?} musicBrainzId - Artist id from musicBrainz api
     * @param {string} name - The artist name from the telegram channel
     * @param {string?} musicBrainzName - The artist name from the musicBrainzApi
     * @param {string?} musicBrainzSortName - The artist sort name from the musicBrainzApi
     * @param {String[]} aliases - Names variation from the musicBrainz
     * @param {String?} gender - Artist gender
     * @param {String[]} genres - Artist music genres
     */
    constructor(name, musicBrainzId, gender, musicBrainzName, musicBrainzSortName, genres = [], aliases = []) {
        this.name = name;
        this.musicBrainzId = musicBrainzId;
        this.aliases = aliases;
        this.gender = gender;
        this.genres = genres;
        this.musicBrainzName = musicBrainzName;
        this.musicBrainzSortName = musicBrainzSortName;
    }

}

module.exports = Artist;

File: api/src/main/java/com/citytube/api/models/Artist.java
package com.citytube.api.models;


import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.Type;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "artist")
public class Artist extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String musicBrainzId;

    private String musicBrainzSortName;

    private String musicBrainzName;

    private String gender;

    @Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private List<String> aliases;

    @Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private List<String> genres;

    @OneToMany(mappedBy = "artist", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Album> albums;


}
